### PR TITLE
Deduplicate paths between old and new PYTHONPATH

### PR DESCRIPTION
**Pull request by ericsong**
Resolves https://github.com/bazelbuild/bazel/issues/15649 by deduplicating the PYTHONPATH after the old and new one has been joined.

### CODE REVIEW

**Comment by ericsong** [comment]
Not sure how to run rerun the `cla/google` check but I have since signed the CLA (with the correct email this time).

**Comment by ckolli5** [comment]
Hello @ericsong, I rerun the cla/google check and it is passed. Thanks!

**Comment by comius** [comment]
I have a question. Is there a way to do this deduplication one level higher, that is when stub parameters are being substituted? Bazel has a lot of machinery to do this, and it might be nicer if the duplicates never get into the template.

**Comment by ericsong** [comment]
I haven't tried it out yet but with that approach, we would be swapping it out [here](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/BazelPythonSemantics.java#L175) ?

I'm not sure if that would be possible since the duplicates are between that and the PYTHONPATH at runtime. We wouldn't know what [old_python_path](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt#L304) would be when the python script is being generated.

**Comment by ericsong** [comment]
@comius thoughts on the above?

File: src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt [hunk 1]
```diff
@@ -298,14 +298,16 @@ def Main():
...
+    for d in python_path_entries
   ]
 
   old_python_path = os.environ.get('PYTHONPATH')
   python_path = os.pathsep.join(python_path_entries)
   if old_python_path:
     python_path += os.pathsep + old_python_path
 
+  python_path = os.pathsep.join(Deduplicate(python_path.split(os.pathsep)))
+
```
Comment (comius) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt:310]: Current implementation with joining, splitting, joining, is not the most readable.

Perhaps rewrite this block, by adding old_python_path to python_path_entries and then deduplicate and join with path separator.

**Review by comius — CHANGES_REQUESTED** [review]

**Comment by comius** [comment]
> I haven't tried it out yet but with that approach, we would be swapping it out [here](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/BazelPythonSemantics.java#L175) ?
> 
> I'm not sure if that would be possible since the duplicates are between that and the PYTHONPATH at runtime. We wouldn't know what [old_python_path](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt#L304) would be when the python script is being generated.

Thanks @ericsong. Since it's coming from the environment, it in fact isn't possible to move it one layer higher.

**Comment by ericsong** [comment]
thanks for the review @comius . I made some changes per your comment.

**Review by comius — APPROVED** [review]

**Pull request closed without merging**